lib/CMakeLists.txt

```diff
@@ -55,12 +55,14 @@ add_llvm_loadable_module(Svf ${SOURCES})
 add_llvm_Library(LLVMSvf ${SOURCES})
 
 link_directories( ${CMAKE_BINARY_DIR}/lib/Cudd )
-llvm_map_components_to_libnames(llvm_libs bitwriter core ipo irreader instcombine instrumentation target linker analysis scalaropts support )
-target_link_libraries(LLVMSvf ${llvm_libs})
+
+## we don't link libs for static library since we rely on target linkage
+# llvm_map_components_to_libnames(llvm_libs bitwriter core ipo irreader instcombine instrumentation target linker analysis scalaropts support )
+# target_link_libraries(LLVMSvf ${llvm_libs})
+
+## we don't link shared library (module) since it will be resolved by opt
 if ( CMAKE_SYSTEM_NAME MATCHES "Darwin")
```

Comment:
Did you handle both OSX and ubuntu system?

Reply:
Only Ubuntu 18.04. I've no OSX at hand right now.